roles/openshift-route-status/tasks/main.yml

```diff
@@ -1,31 +1,26 @@
 ---
 
-- name: "Lookup route url for {{ route }}"
-  shell: "oc get route {{ route }} -o jsonpath='{ .spec.host }'"
-  register: url
-  when:
-  - route is defined
-  - route|trim != ''
+- name: Verify Required Parameters
+  fail:
+    msg: "'route', 'protocol' and 'status' Parameters Must be Provided!"
+  when: route is not defined or route|trim == '' or protocol is not defined or protocol|trim == '' or status is not defined or status|trim == ''
 
-- name: "Set default delay time"
+- name: "Set the target namespace option if supplied"
   set_fact:
-    delay: "{{ delay | default(5) }}"
+    target_namespace: "-n {{ namespace }}"
```

Comment:
Can you please model this set_fact after how it's done in some of these other roles - i.e.: only pre-pend `-n` if not already supplied? Also, even though the roles aren't meant to be openshift-applier specific, we often use `target_namespace` as the fact name to allow for a easier use in the openshift-applier. 
https://github.com/redhat-cop/casl-ansible/blob/master/roles/openshift-labels/tasks/main.yml#L9